### ISSUE TITLE
Warning is not reported for a type shadowing a declaration from base interface.

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#38711]
```

        [Fact]
        public void NestedTypes_33()
        {
            var source1 =
@"
interface I1
{
    interface I2
    { }
}

interface I3 : I1
{
    interface I2
    {
    }
}

class C1
{
    public interface I4
    { }
}

class C3 : C1
{
    public interface I4
    {
    }
}
";

            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
                                                 parseOptions: TestOptions.Regular);

            compilation1.VerifyDiagnostics(
                // (23,22): warning CS0108: 'C3.I4' hides inherited member 'C1.I4'. Use the new keyword if hiding was intended.
                //     public interface I4
                Diagnostic(ErrorCode.WRN_NewRequired, "I4").WithArguments("C3.I4", "C1.I4").WithLocation(23, 22)
                );
        }
```
